indexing-service/src/main/java/org/apache/druid/indexing/overlord/http/OverlordResource.java

```diff
@@ -225,18 +222,29 @@ public Response taskPost(final Task task, @Context final HttpServletRequest req)
         taskQueue -> {
           try {
             taskQueue.add(task);
+
+            auditManager.doAudit(
```

Comment:
can we filter task submissions that were initiated by the system itself? E.g. compaction duty, MSQ controller task, etc?

Reply:
Yes, this is already happening inside the respective audit impls, such as `SQLAuditManager` and `LoggingAuditManager`. We check if an audited event was started by an internal service (by checking the author field) and then audit that event only if system event auditing is enabled via config (false by default).

This is better than checking at each call site if the request was initiated by an internal service.